### PR TITLE
refresh config/services and new  migration ForeignKeyInSubscriptions

### PR DESCRIPTION
**Pull request by federfelipe**

### CODE REVIEW

File: app/Data/Repositories/Data.php [hunk 1]
```diff
@@ -135,15 +140,21 @@ public function getClipping($year)
     public function createSubscription($input)
     {
         $model = new Subscription();
+        $this->usersRepository = new UsersRepository();
```
Comment (antonioribeiro) [inline app/Data/Repositories/Data.php]: Use a local var instead of property

File: app/Data/Repositories/Data.php [hunk 2]
```diff
@@ -135,15 +140,21 @@ public function getClipping($year)
...
 
-        $input = $input->only($model->getFillable());
+        if (!$user = $this->usersRepository->findByBirthdateAndRegistration($input->input('birthdate'), $input->input('registration')))
+        {
```
Comment (antonioribeiro) [inline app/Data/Repositories/Data.php]: This is too complex, please create a method to find the user.

File: app/Http/Controllers/SocialAuthController.php [hunk 3]
```diff
@@ -16,15 +16,27 @@ public function __construct(SocialUserService $socialUserService)
 
     public function redirect($socialNetwork)
     {
+        session_start();
```
Comment (antonioribeiro) [inline app/Http/Controllers/SocialAuthController.php]: Use Laravel

File: app/Http/Controllers/SocialAuthController.php [hunk 4]
```diff
@@ -16,15 +16,27 @@ public function __construct(SocialUserService $socialUserService)
...
+       public function socialNetworkCallback($socialNetwork)
+    {
+        session_start();
+        $regBirth = ["registration"=> $_SESSION['registration'] , "birthdate" => $_SESSION['birthdate']];
+
+        if (!$this->socialUserService->find($socialNetwork, $this->getDriver($socialNetwork)->user(), $regBirth)) {
             return redirect()->route('login');
         }
-
+        dd('AQUI tá quase lá');
```
Comment (antonioribeiro) [inline app/Http/Controllers/SocialAuthController.php]: Remove this

File: app/Repositories/SocialUserRepository.php [hunk 5]
```diff
@@ -40,18 +40,15 @@ public function createUser($email, $socialUser)
...
         $userModel->email = $email;
         $userModel->password = 'Empty';
-        $userModel->uf = 'RJ';
-        $userModel->role_id = '99';
-        $uuid = Uuid::uuid4();
-        $userModel->uuid = $uuid;
-        $userModel->save();
+        $userModel->registration = $regBirth['registration'];
+        $userModel->birthdate = $regBirth['birthdate'];
+      //  $userModel->uf = 'RJ';
```
Comment (antonioribeiro) [inline app/Repositories/SocialUserRepository.php]: Remove those comments or comment those comments

File: app/Data/Repositories/Data.php (hunk 2 above)
Comment (federfelipe) [inline app/Data/Repositories/Data.php]: ok

File: app/Http/Controllers/SocialAuthController.php (hunk 3 above)
Comment (federfelipe) [inline app/Http/Controllers/SocialAuthController.php]: ok

File: app/Http/Controllers/SocialAuthController.php (hunk 4 above)
Comment (federfelipe) [inline app/Http/Controllers/SocialAuthController.php]: ok

File: app/Repositories/SocialUserRepository.php (hunk 5 above)
Comment (federfelipe) [inline app/Repositories/SocialUserRepository.php]: ok

**Pull request closed without merging**